src/test/java/com/algorand/algosdk/transaction/TestTransaction.java

```diff
@@ -655,4 +655,26 @@ public void testTransactionWithLease() throws Exception {
         assertThat(encodedOutBytes).isEqualTo(goldenString);
         TestUtil.serializeDeserializeCheck(stx);
     }
+
+    @Test
+    public void EmptyByteArraysShouldBeRejected() throws Exception {
```

Comment:
Not sure that the semantics of empty string and null byte array are the same.

Reply:
These fields don't accept a string, so "empty string" doesn't apply to the underlying problem.